flytekit/core/data_persistence.py

```diff
@@ -106,28 +141,15 @@ def data_config(self) -> DataConfig:
 
     def get_filesystem(
         self, protocol: typing.Optional[str] = None, anonymous: bool = False, **kwargs
-    ) -> typing.Optional[fsspec.AbstractFileSystem]:
+    ) -> fsspec.AbstractFileSystem:
         if not protocol:
             return self._default_remote
-        if protocol == "file":
-            kwargs["auto_mkdir"] = True
-        elif protocol == "s3":
-            s3kwargs = s3_setup_args(self._data_config.s3, anonymous=anonymous)
-            s3kwargs.update(kwargs)
-            return fsspec.filesystem(protocol, **s3kwargs)  # type: ignore
-        elif protocol == "gs":
-            if anonymous:
-                kwargs["token"] = _ANON
-            return fsspec.filesystem(protocol, **kwargs)  # type: ignore
-        elif protocol == "abfs":
-            kwargs["anon"] = False
-            return fsspec.filesystem(protocol, **kwargs)  # type: ignore
-
-        # Preserve old behavior of returning None for file systems that don't have an explicit anonymous option.
```

Comment:
From the type hint on `get_filesystem_for_path` it looks like we assume `get_filesystem` never returns `None` so I thought probably best to delete this. If anyone thinks its important I'm happy to add it back.

Reply:
@wild-endeavor I think you wanted to delete this already? So this seems like a good idea